### ISSUE TITLE
The --forceall flag is not propagated to subworkflows

### ISSUE THREAD

**Issue by dkuzminov** [snakemake/snakemake#86]
I'm trying to run a workflow with subworkflows with the `--forceall` flag specified.

Expected behavior: subworkflow rebuild all rules.

Actual behavior: subworkflows don’t rebuild any rule if the files are present.

‌

As I can see, the problem is in the `snakemake` function in the `__init__` file, line 414. The subsnakemake variable is being created as a partial function where the `forceall` flag is not specified (and the default value = `False` is used later). Actually the whole list of frozen parameters should be revised as there should be other parameters that need to be propagated to subworkflows.

**Comment by johanneskoester** [comment]
Yes, this was in a sense intentional. There is a usecase for subworkflows where you see the current workflow as a downstream analysis. In such a case, you typically do not want forceall to propagate to subworkflows. However, I also think that your usecase is valid. Let us consider this an enhancement. One could add a flag in the subworkflow syntax that switches between the two modes.

**Comment by m-jahn** [comment]
@schnea can this be closed as sub-workflows are already deprecated since v8.0.0?

**Comment by schnea** [comment]
Sure, missed that subworkflows were already deprecated
